sparse/core.py

```diff
@@ -1121,6 +1121,26 @@ def conjugate(self, out=None):
         assert out is None
         return self.elemwise(np.conjugate)
 
+    def triu(self, k=0):
+        assert self.ndim >= 2
+
+        mask = self.coords[-2] + k <= self.coords[-1]
+
+        coords = self.coords[:, mask]
+        data = self.data[mask]
+
+        return COO(coords, data, self.shape, self.has_duplicates, self.sorted)
+
+    def tril(self, k=0):
+        assert self.ndim >= 2
```

Comment:
We should probably change these to `ValueError`s with informative error messages.  A brief test using `pytest.raises` might also be useful.

Reply:
If we follow Numpy strictly strictly, `ndim == 1` works. Not sure if we should support it here, since it is an artefact of the way it is implemented in Numpy rather than an actual feature. I haven't tested scalars (`ndim == 0`) yet. 

I want to mimic numpy behaviour except, of course artefacts. I guess your approach is good but I'd hold off on the test, simply because we're not mimicking Numpy.

Reply:
In that case then we should raise a `NotImplementedError`.  Using `assert` in user code is generally avoided.  For example when people run Python in optimized mode `assert` statements get passed over.  Generally the convention is to always `raise ValueError("informative message")` or `raise NotImplementedError("informative message")` instead of using `assert`.